app/auth-portal/src/content/tutorial/08-asset_authoring.md

```diff
@@ -0,0 +1,442 @@
+the qualification is ready for us to use.
+
+### Authoring Action Functions
+
+Action functions are a way of being able to attach behavior to our components. The types of actions we can do are:
+
+* Create
+* Delete
+* Refresh
+
+Let's start with authoring create, delete and refresh functions. It's important to set the `entrypoint` of the function attributes
+to match the name of the action.
+
+![create-action.png](/tutorial-img/08-asset-authoring/create-action.png)
+
+```js
+async function create(component: Input): Promise<Output> {
+  if (component.properties.resource?.payload) {
+    return {
+      status: "error",
+      message: "Resource already exists",
+      payload: component.properties.resource.payload,
+    };
+  }
+  
+  const child = await siExec.waitUntilEnd("aws", [
+    "ec2",
+    "create-volume",
+    "--region",
+    component.properties.domain?.region,
+    "--cli-input-json",
+    component.properties.code["awsEc2EbsVolumeJSON"]?.code,
+  ]);
+
+  if (child.exitCode !== 0) {
+    console.error(child.stderr);
+    return {
+      status: "error",
+      message: `Unable to create EBS Volume, AWS CLI 2 exited with non zero code: ${child.exitCode}`,
+    };
+  }
+
+  return { payload: JSON.parse(child.stdout), status: "ok" };
+}
+```
+
+![delete-func.png](/tutorial-img/08-asset-authoring/delete-func.png)
+
+```js
+async function deleteResource(component: Input): Promise<Output> {
+  const resource = component.properties.resource?.payload;
+
+  const child = await siExec.waitUntilEnd("aws", [
+    "ec2",
+    "delete-volume",
+    "--region",
+    component.properties.domain.region,
+    "--volume-id",
+    resource.VolumeId,
+  ]);
+
+  if (child.exitCode !== 0) {
+    console.error(child.stderr);
+    return {
+        status: "error",
+        payload: resource,
+        message: `Unable to delete E, AWS CLI 2 exited with non zero code: ${child.exitCode}`,
+      };
+  }
+
+  return { payload: null, status: "ok" };
+}
+```
+
+![refresh-func.png](/tutorial-img/08-asset-authoring/refresh-func.png)
+
+```js
+async function refresh(component: Input): Promise<Output> {
+  const resource = component.properties.resource?.payload;
+  if (!resource) {
+    return {
+      status: component.properties.resource?.status ?? "ok",
+      message: component.properties.resource?.message,
+    };
+  }
+
+  const child = await siExec.waitUntilEnd("aws", [
+    "ec2",
+    "describe-volumes",
+    "--volume-ids",
+    resource.VolumeId,
+    "--region",
+    component.properties.domain.region,
+  ]);
+
+  if (child.exitCode !== 0) {
+    console.log(`Volume ID: ${resource.VolumeId}`);
+    console.error(child.stderr);
+    return {
+      payload: resource,
+      status: "error",
+      message: `AWS CLI 2 "aws ec2 describe-volumes" returned non zero exit code (${child.exitCode})`,
+    };
+  }
+
+  return { payload: JSON.parse(child.stdout).Volumes[0], status: "ok" };
+}
+```
+
```

Comment:
Suggest you add some explanation, walking the user through what is in the code above and how you built it.